src/okta/services/okta.service.ts

```diff
@@ -10,226 +10,93 @@
  * See the License for the specific language governing permissions and limitations under the License.
  */
 
-import { Inject, Injectable, Injector } from '@angular/core';
+import { Inject, Injectable } from '@angular/core';
 import {
   assertIssuer,
   assertClientId,
   assertRedirectUri,
 } from '@okta/configuration-validation';
 
-import { OKTA_CONFIG, OktaConfig, AuthRequiredFunction } from '../models/okta.config';
-import { UserClaims } from '../models/user-claims';
-import { TokenManager, AccessToken, IDToken } from '../models/token-manager';
+import { OKTA_CONFIG, OktaConfig } from '../models/okta.config';
 
 // eslint-disable-next-line node/no-unpublished-import
 import packageInfo from '../packageInfo';
 
 /**
  * Import the okta-auth-js library
  */
-import OktaAuth from '@okta/okta-auth-js';
+import { OktaAuth, AuthState } from '@okta/okta-auth-js';
 import { Observable, Observer } from 'rxjs';
 
+/**
+ * Scrub scopes to ensure 'openid' is included
+ * @param scopes
+ */
+function scrubScopes(scopes: string[]): void {
+  if (scopes.indexOf('openid') >= 0) {
+    return;
+  }
+  scopes.unshift('openid');
+}
+
 @Injectable()
-export class OktaAuthService {
-    private oktaAuth: OktaAuth;
+export class OktaAuthService extends OktaAuth {
     private config: OktaConfig;
     private observers: Observer<boolean>[];
+
     $authenticationState: Observable<boolean>;
 
-    constructor(@Inject(OKTA_CONFIG) config: OktaConfig, private injector: Injector) {
-      this.observers = [];
+    constructor(@Inject(OKTA_CONFIG) config: OktaConfig) {
+      const originalConfig = config;
+      config = Object.assign({
+        transformAuthState: async (oktaAuth: OktaAuth, authState: AuthState) => {
+          if (config.isAuthenticated) {
+            authState.isAuthenticated = await config.isAuthenticated(this);
+          }
+          if (originalConfig.transformAuthState) {
+            authState = await originalConfig.transformAuthState(oktaAuth, authState);
+          }
+          return authState;
+        }
+      }, config);
```

Comment:
how about add default option for `restoreOriginalUri` to use angular router to make in-page redirect?

Reply:
is there a better user experience using the router, vs. window.location?

Reply:
If I understand the `window.location.replace/assign` correctly, it will reload the page. But the angular router uses the `history` API which  still maintains the SPA context.

Reply:
So no default `restoreOriginalUri ` implementation for angular?

Reply:
added